Slim/Control/Queries.pm

```diff
@@ -5010,18 +5016,26 @@ sub _songData {
 			$remoteMeta->{T} = $remoteMeta->{samplerate};
 			$remoteMeta->{I} = $remoteMeta->{samplesize};
 			$remoteMeta->{W} = $remoteMeta->{releasetype};
+
```

Comment:
I would not do it like that b/c $song->isLive is only set if there is no HTTP content-length (direct or proxied) so it will not be set if you have a protocol handler because it might not use Slim::Player::Protocol::HTTP at all (not descend from it) or set a CL. So you'd then block the 'V' parameter. 
If you want to not have the PH have a isLive method, which is fine, then I'd rather say:
```
$remoteData->{V} = $remoteMeta->{live_edge} // ($song && $song->isLive ? 0 : -1);
```

Or anything that is more readbable. I'm sure @mherger will laugh seeing that line, because he knows I tend to do things compact to a point they become unreadable. But the idea is V is live_meta is they exists (precedence), otherwise is 0 is isLive and -1 otherwise.

Reply:
It's totally fine and doesn't use pointers 😀. But I always have to double check what version `//` was introduced with. It's an operator I still don't use... I only found it in one place in LMS, a change in the HTTP code you did 😉.

Reply:
> I would not do it like that b/c $song->isLive is only set if there is no HTTP content-length (direct or proxied) so it will not be set if you have a protocol handler because it might not use Slim::Player::Protocol::HTTP at all (not descend from it) or set a CL. So you'd then block the 'V' parameter. If you want to not have the PH have a isLive method, which is fine, then I'd rather say:
> 
> ```
> $remoteData->{V} = $remoteMeta->{live_edge} // ($song && $song->isLive ? 0 : -1);
> ```
> 
> Or anything that is more readable. I'm sure @mherger will laugh seeing that line, because he knows I tend to do things compact to a point they become unreadable. But the idea is V is live_meta is they exists (precedence), otherwise is 0 is isLive and -1 otherwise.

Yes, I think if a ProtocolHandler needs to support the isLive behaviour it needs to populate song->isLive.  As that is used elsewhere in LMS.  So no need for a PH isLive method.

I'll update to your compact suggestion!

I'm definitely going to have up my perl game if I'm going to have a less bumpy ride on future pull requests!😀

Reply:
I can be a pain sometimes and I'm (very) far from being always right, so feel free to ignore my comments 😃